core/src/main/scala/scalaz/LazyEitherT.scala

```diff
@@ -105,7 +105,7 @@ object LazyEitherT extends LazyEitherTInstances {
     u1: Unapply[Functor, FAB]{type A = AB},
     u2: Unapply2[Bifunctor, AB]{type A = A0; type B = B0},
     l: Leibniz.===[AB, LazyEither[A0, B0]]
-  ): LazyEitherT[u1.M, A0, B0] = LazyEitherT(l.subst[u1.M](u1(fab)))
+  ): LazyEitherT[u1.M, u2.A, u2.B] = LazyEitherT(l.subst[u1.M](u1(fab)))
```

Comment:
It is a problem for `u1.M`, too, but we can't fix it due to scala/bug#5075. (This is *not* scala/bug#2712, despite some references therein and surface similarities.)

We don't have that problem for `A0` and `B0`, so we can Do The Right Thing for those.